xgcm/test/test_grid_ufunc.py

```diff
@@ -441,6 +454,198 @@ def grad_to_inner(a):
+
+        expected = a + 1
+        result = apply_as_grid_ufunc(
+            increment,
+            a,
+            axis=[("depth",)],
+            grid=grid,
+            signature="(X:left)->(X:left)",
+            boundary_width=None,
+            dask="allowed",
+            map_overlap=True,
+        ).compute()
+        assert_equal(result, expected)
+
+        # in this case the result should be the same as using just map_blocks
+        expected_data = dask.array.map_blocks(increment, a.data)
+        np.testing.assert_equal(result.data, expected_data)
+
+    @pytest.mark.skip
+    def test_only_some_core_dims_are_chunked(self):
+        raise NotImplementedError
+
+    def test_ufunc_changes_chunksize(self):
+        @as_grid_ufunc(
+            "(X:outer)->(X:center)",
+            boundary_width={"X": (1, 0)},
+            dask="allowed",
+            map_overlap=True,
+        )
+        def diff_outer_to_center(a):
+            """Mocking up a function which can only act on in-memory arrays, and requires no padding"""
+            if isinstance(a, np.ndarray):
+                return a[..., 1:] - a[..., :-1]
+            else:
+                raise TypeError
+
+        grid = create_1d_test_grid("depth")
+        da = np.sin(grid._ds.depth_o * 2 * np.pi / 9).chunk(3)
+        da.coords["depth_o"] = grid._ds.depth_o
+
+        with pytest.raises(
+            NotImplementedError, match="includes one of the axis positions"
+        ):
+            diff_outer_to_center(
+                grid,
+                da,
+                axis=[("depth",)],
+            ).compute()
+
+    def test_multiple_inputs(self):
+        @as_grid_ufunc(
+            "(X:left),(X:right)->(X:center)",
+            boundary_width=None,
+            map_overlap=True,
+            dask="allowed",
+        )
+        def multiply_left_right(a, b):
+            """Mocking up a function which can only act on in-memory arrays, and requires no padding"""
+            if isinstance(a, np.ndarray) and isinstance(b, np.ndarray):
+                return np.multiply(a, b)
+            else:
+                raise TypeError
+
+        grid = create_1d_test_grid("depth")
+        a = np.sin(grid._ds.depth_g * 2 * np.pi / 9).chunk(2)
+        a.coords["depth_g"] = grid._ds.depth_g
+        b = np.cos(grid._ds.depth_r * 2 * np.pi / 9).chunk(2)
+        b.coords["depth_r"] = grid._ds.depth_r
+
+        depth_c_coord = xr.DataArray(np.arange(1, 10), dims="depth_c")
+        expected = xr.DataArray(
+            np.multiply(a.data, b.data),
+            dims=["depth_c"],
+            coords={"depth_c": depth_c_coord},
+        )
+
+        result = multiply_left_right(grid, a, b, axis=[("depth",), ("depth",)])
+        assert_equal(result, expected)
+
+    def test_multiple_outputs(self):
+        def diff_center_to_inner(a, axis):
+            result = a - np.roll(a, shift=1, axis=axis)
+            return np.delete(result, 0, axis)  # remove first element along axis
```

Comment:
It would be more efficient to do something like `indexer = (slice(None),) * (axis-1) + (slice(0, -1),)`, but given that this is just a single test, it doesn't seem worth bothering.